internal/runners/projects/edit.go

```diff
@@ -0,0 +1,179 @@
+	}
+
+	return nil
+}
+
+type Edit struct {
+	auth   *authentication.Auth
+	out    output.Outputer
+	prompt prompt.Prompter
+	config configGetter
+}
+
+func NewEdit(prime primeable) *Edit {
+	return &Edit{
+		auth:   prime.Auth(),
+		out:    prime.Output(),
+		prompt: prime.Prompt(),
+		config: prime.Config(),
+	}
+}
+
+func (e *Edit) Run(params EditParams) error {
+	err := params.validate()
+	if err != nil {
+		return locale.WrapError(err, "err_edit_invalid_params", "Invalid edit parameters")
+	}
+
+	editParams := projects.NewEditProjectParams()
+	editParams.SetOrganizationName(params.Namespace.Owner)
+	editParams.SetProjectName(params.Namespace.Project)
+
+	editMsg := locale.Tl("edit_prompt", "You are about to edit the following fields for the project [NOTICE]{{.V0}}[/RESET]:\n", params.Namespace.String())
+	editable := &mono_models.ProjectEditable{}
+	if params.ProjectName != "" {
+		editMsg += locale.Tl("edit_prompt_name", "  - Name: {{.V0}}\n", params.ProjectName)
+		editable.Name = params.ProjectName
+	}
+
+	if params.Visibility != "" {
+		editMsg += locale.Tl("edit_prompt_visibility", "  - Visibility: {{.V0}}\n", params.Visibility)
+		if strings.EqualFold(params.Visibility, visibilityPublic) {
+			editable.Private = p.BoolP(false)
+		} else {
+			editable.Private = p.BoolP(true)
+		}
+	}
+
+	if params.Repository != "" {
+		editMsg += locale.Tl("edit_prompt_repo", "  - Repository: {{.V0}}\n", params.Repository)
+		editable.RepoURL = &params.Repository
+	}
+	editMsg += locale.Tl("edit_prompt_confirm", "Continue?")
+
+	edit, err := e.prompt.Confirm("", editMsg, p.BoolP(false))
+	if err != nil {
+		return locale.WrapError(err, "err_edit_prompt", "Could not prompt for edit confirmation")
+	}
+
+	if !edit {
+		e.out.Print(locale.Tl("edit_cancelled", "Project edit cancelled"))
+		return nil
+	}
+
+	if err = model.EditProject(params.Namespace.Owner, params.Namespace.Project, editable); err != nil {
+		return locale.WrapError(err, "err_edit_project", "Could not edit project")
+	}
+
+	if err = e.editLocalCheckouts(params); err != nil {
+		return locale.WrapError(err, "err_edit_local_checkouts", "Could not edit local checkouts")
+	}
+
+	if err = e.updateProjectMapping(params); err != nil {
+		return locale.WrapError(err, "err_edit_project_mapping", "Could not update project mapping")
+	}
+
+	e.out.Print(locale.Tl("edit_success", "Project edited successfully"))
+
+	return nil
+}
+
+func (e *Edit) editLocalCheckouts(params EditParams) error {
+	localProjects := projectfile.GetProjectMapping(e.config)
+
+	var localCheckouts []string
+	for namespace, checkouts := range localProjects {
+		if namespace == params.Namespace.String() {
+			localCheckouts = append(localCheckouts, checkouts...)
+		}
+	}
+
+	for _, checkout := range localCheckouts {
+		err := e.editLocalCheckout(params.Namespace.Owner, checkout, params)
+		if err != nil {
+			return errs.Wrap(err, "Could not edit local checkout at %s", checkout)
+		}
+	}
```

Comment:
Is there any reason to not combine these two loops into one that just calls `e.editLocalCheckout()` in the range over `localProjects`?

Reply:
I tend to avoid nesting for loops as I find it more easily readable. However, if you feel strongly about it I can nest them.

Reply:
Just so we're on the same page,

```
	for namespace, checkouts := range localProjects {
		if namespace == params.Namespace.String() {
			err := e.editLocalCheckout(params.Namespace.Owner, checkout, params)
			if err != nil {
				return errs.Wrap(err, "Could not edit local checkout at %s", checkout)
			}
		}
	}
```

is what you mean by "nesting for loops"?

Reply:
Ohhhhh wait. I totally missed the trailing `...` in your append. Forget it. Its fine the way you have it.